src/handlers/mozilla-auth0.js

```diff
@@ -95,61 +104,98 @@ class Handler {
       return;
     }
 
-    let a0 = await this.getManagementApi();
-    let profile = await new Promise((resolve, reject) =>
-      a0.getUser(req.user.sub, (err, prof) => err ? reject(err) : resolve(prof)));
+    const profile = await this.profileFromIdentity(req.user.sub);
 
     if ('active' in profile && !profile.active) {
       debug('user is not active; rejecting');
       return;
     }
 
-    let user = this.userFromProfile(profile);
+    const user = this.userFromProfile(profile);
     user.expires = new Date(req.user.exp * 1000);
+
     return user;
   }
 
-  userFromProfile(profile) {
-    let user = new User();
-
-    // we recognize a few different kinds of 'identities' that auth0 can send
-    // our way.  Each of these translates into a different identityProviderId,
-    // which authorizers will later use to figure out what scopes this user
-    // has.  We do not ever expect to have more than one identity in this array,
-    // in a practical sense.
-    for (let {provider, connection} of profile.identities) {
-      // The 'Mozilla-LDAP' connection corresponds to an LDAP login. For this
-      // login, emails are a unique identifier
-      if (provider === 'ad' && connection === 'Mozilla-LDAP') {
-        if (profile.email_verified) {
-          user.identity = 'mozilla-ldap/' + profile.email;
-          break;
-        }
+  isIdentityProviderRecognized({provider, connection}) {
+    if (
+      provider === 'ad' && connection === 'Mozilla-LDAP' ||
       // The 'email' connection corresponds to a passwordless login.
-      } else if (provider === 'email' && connection === 'email') {
-        if (profile.email_verified) {
-          user.identity = 'email/' + profile.email;
-          break;
-        }
-      // Login with google really only gives us an email, too.
-      } else if (provider === 'google-oauth2' && connection === 'google-oauth2') {
-        if (profile.email_verified) {
-          user.identity = 'email/' + profile.email;
-          break;
+      provider === 'email' && connection === 'email' ||
+      provider === 'google-oauth2' && connection === 'google-oauth2' ||
+      provider === 'github' && connection === 'github'
+    ) {
+      return true;
+    }
+
+    return false;
+  }
+
+  async userFromIdentity(urlEncodedIdentity) {
+    const profile = await this.profileFromIdentity(urlEncodedIdentity);
```

Comment:
doesn't this need to be urldecoded first?

Reply:
Right now the decoding happens first thing in `profileFromIdentity`. I could always decode it first before calling `profileFromIdentity`. I'll do that to make things clearer.